lucene/core/src/java/org/apache/lucene/search/SortField.java

```diff
@@ -394,6 +399,21 @@ public String toString() {
     return buffer.toString();
   }
 
+
+  /**
+   * For numeric sort fields, setting this field, indicates that
+   * the same numeric data has been indexed with two fields: doc values and points and
+   * that these fields have the same name.
+   * This allows to use sort optimization and skip non-competitive documents.
+   */
+  public void setSkipNonCompetitiveDocs() {
```

Comment:
I wonder if we should make the option less ambiguous by naming it something like `setCanUsePoints` or something along those lines ? I am afraid that this could be misleading if we set this option on a sort field that use the _score or a keyword field for instance. We should maybe at least validate that the field is not `null` ?

Reply:
How about `setCanUsePointsForSort`? I am also fine with `setCanUsePoints`.

Reply:
@jpountz  Do you have suggestions how we should expose sort optimization in 8.x? Are you ok with a proposal to have  a method for `SortField::setCanUsePointsForSort` that when set `true` indicates that the same data is used for indexing points and docValues?

Reply:
IMO `setCanUsePoints` is good enough since this is a "SortField" so it should be obvious that this is for sorting?